src/main/scala/org/scalameter/Executor.scala

```diff
@@ -1,18 +1,21 @@
 package org.scalameter
 
 
-import collection._
-import compat._
-import utils.{withGCNotification, Tree}
+import utils.Tree
 
 
 
 trait Executor {
 
+  def measurer: Measurer
+
   def run[T](setuptree: Tree[Setup[T]], reporter: Reporter, persistor: Persistor): Tree[CurveData] = {
+    val context = measurer.prepareContext(setuptree.context)
```

Comment:
It seems to me that the initial `prepareContext` call must:
a) do some side-effecting actions to prepare a testing environment
b) return a couple of values that describe the prepared testing environment

Initially, I was hoping we could replace the top-level context altogether before it gets used to create `Tree[Setup[T]]`, and that `BasePerformanceTest#defaultConfig` could be used for this purpose. Now I see that this is broken - `defaultConfig` is just ignored.

The way to fix it is to use `defaultConfig` in `BasePerformanceTest` to initialize the `setupzipper` dynamic variable with the correct context.
Something like this should do it:

    def defaultConfig = Context.empty // this is user-overridable
    final def initialConfig = measurer.prepareContext(currentContext ++ Context.empty)
    setupzipper.value = initialConfig

This should ensure that `Tree` objects have the correct contexts while the test tree is being constructed.
Eventually, it will bring the measurer-specific parameters to each and every test snippet.

The problem here is, of course, that contexts in the end contain things like the lookup table, which should not be persisted. I think the solution is to introduce transient keys, which we never persist. We seem to be hitting a fair amount of such keys anyway, and it's awkward to tweak the API in an effort to avoid them being serialized.